src/main/java/com/amazon/opendistroforelasticsearch/sql/antlr/StringSimilarity.java

```diff
@@ -0,0 +1,57 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.antlr;
+
+import org.apache.lucene.search.spell.LevenshteinDistance;
+import org.apache.lucene.search.spell.StringDistance;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * String similarity for similar string(s) computation
+ */
+public class StringSimilarity {
+
+    private final StringDistance algorithm = new LevenshteinDistance();
+
+    private final Collection<String> candidates;
+
+    public StringSimilarity(Collection<String> candidates) {
+        this.candidates = Collections.unmodifiableCollection(candidates);
+    }
+
+    /**
+     * Calculate similarity distance between target and candidate strings.
```

Comment:
I think that everything here is actively deceiving reader :(

1. The javadoc says that the method calculates similartiy distance. In reality it returns string the most similar to supplied
2. The javadoc says "one or more most similar), but it will always return only one, last string that is most similar (e.g. if there will be 2 strings with the same distance, algorithm will still return only one string

Reply:
Good catch! Update all confusing JavaDoc. Thanks!